### DISCUSSION TITLE
Adding a New Objective Function to mHM

### DISCUSSION THREAD

**Discussion by ViniciusBMoreira** [mhm-ufz/mHM#80] [Q&A] [answered]
Hello everyone!

I'm planning to add a new objective function to mHM. I understand that the simplest way would be to edit an existing one and then compile the model. So far, I've been manipulating the file mhm\src\mHM\mo_objective_function.F90, as my intention is to use spatial metrics and flow for calibrating the model.

Is there any other step that needs to be taken to complete my goal?

Best regards,
Bryan

**Comment by pallavhydro** [comment, marked as answer]
Hi Bryan

Using an existing subroutine/ function for a new objective function - now that's one hacky idea! :) 

Yes mo_objective_function.F90 is the file you need to work on. Just follow the structure in this file and that should be enough. 

Just keep in mind that if you use git to keep your repository up-to-date then this approach can lead to complexities... but I guess you have already foreseen this.


Good luck
Pallav

**Reply by ViniciusBMoreira** [reply]
thanks a lot!

Bryan

**Reply by pallavhydro** [reply]
Hi @BryanSouzaM 

Great. Don't forget to mark the answer to this discussion ;)

Cheers
P
